File: backend/app/schemas/social.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List


class SocialAccountCreate(BaseModel):
    provider: str = Field(..., max_length=50, description="Platform: wechat / dingtalk / feishu")
    account_name: Optional[str] = Field(None, max_length=200, description="Optional account/enterprise name")


class SocialAccountUpdate(BaseModel):
    account_name: Optional[str] = Field(None, max_length=200)
    status: Optional[str] = Field(None, max_length=50)


class SocialAccountResponse(BaseModel):
    id: str
    user_id: str
    provider: str
    account_name: Optional[str]
    connection_type: str
    sync_status: str
    last_sync_at: Optional[datetime]
    last_error: Optional[str]
    sync_count: int
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SocialMessageResponse(BaseModel):
    id: str
    user_id: str
    account_id: str
    platform: str
    conversation_id: Optional[str]
    conversation_name: Optional[str]
    message_uid: str
    sender_name: Optional[str]
    sender_id: Optional[str]
    content_raw: Optional[str]
    content_text: Optional[str]
    message_type: str
    attachments: Optional[str]
    sent_at: Optional[datetime]
    is_me: bool
    status: str
    knowledge_id: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SocialUploadResult(BaseModel):
    success: bool
    parsed_count: int
    skipped_count: int
    error: Optional[str] = None


class SocialSaveToKnowledgeRequest(BaseModel):
    tag_ids: Optional[List[str]] = Field(None, description="Optional tag IDs or names to attach")
    brain_side: Optional[str] = Field("network", description="Target brain side: personal / network / both")

File: frontend/src/api/social.ts
import api from './client';

export interface SocialAccount {
  id: string;
  user_id: string;
  provider: 'wechat' | 'dingtalk' | 'feishu';
  account_name: string | null;
  connection_type: string;
  sync_status: string;
  last_sync_at: string | null;
  last_error: string | null;
  sync_count: number;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface SocialAccountCreateData {
  provider: 'wechat' | 'dingtalk' | 'feishu';
  account_name?: string;
}

export interface SocialMessage {
  id: string;
  user_id: string;
  account_id: string;
  platform: string;
  conversation_id: string | null;
  conversation_name: string | null;
  message_uid: string;
  sender_name: string | null;
  sender_id: string | null;
  content_raw: string | null;
  content_text: string | null;
  message_type: string;
  attachments: string | null;
  sent_at: string | null;
  is_me: boolean;
  status: string;
  knowledge_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface SocialUploadResult {
  success: boolean;
  parsed_count: number;
  skipped_count: number;
  error?: string;
}

export const socialApi = {
  listAccounts: () => api.get<SocialAccount[]>('/api/v1/social/accounts'),
  createAccount: (data: SocialAccountCreateData) =>
    api.post<SocialAccount>('/api/v1/social/accounts', data),
  deleteAccount: (id: string) => api.delete(`/api/v1/social/accounts/${id}`),
  uploadFile: (id: string, file: File) => {
    const formData = new FormData();
    formData.append('file', file);
    return api.post<SocialUploadResult>(`/api/v1/social/accounts/${id}/upload`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
  },
  listMessages: (params?: {
    account_id?: string;
    conversation_id?: string;
    q?: string;
    skip?: number;
    limit?: number;
  }) => api.get<SocialMessage[]>('/api/v1/social/messages', { params }),
  getMessage: (id: string) => api.get<SocialMessage>(`/api/v1/social/messages/${id}`),
  saveToKnowledge: (id: string, tag_ids?: string[], brain_side?: string) =>
    api.post(`/api/v1/social/messages/${id}/save-to-knowledge`, { tag_ids, brain_side }),
  deleteMessage: (id: string) => api.delete(`/api/v1/social/messages/${id}`),
};
